src/order/extension/well.lean

```diff
@@ -11,6 +11,12 @@ import set_theory.ordinal.arithmetic
 
 This file constructs a well-order (linear well-founded order) which is an extension of a given
 well-founded order.
+
+## Proof idea
+
+Because our order `r : α → α → Prop` is well-founded, we can define its *rank* function
+`rank : α → ordinal`. We can also find a (not necessarily well-founded) extension `s` of `r`. Then
+we can define `a < b` as `rank a < rank b ∨ (rank a = rank b ∧ s a b)`.
```

Comment:
Where did you use a non-well-founded extension? Instead, we use a well-ordering on `α` that is not necessarily an extension of `r`.